platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -1493,7 +1493,8 @@ def action_on_sw_control(cls, sfp):
     @classmethod
     def action_on_fw_control(cls, sfp):
         logger.log_info(f'SFP {sfp.sdk_index} is set to firmware control')
-        sfp.set_control_type(SFP_FW_CONTROL)
+        if sfp.get_control_type() != SFP_FW_CONTROL:
```

Comment:
if it is already in fw control why do we need the log in notice? maybe you should use debug as already in fw control?

Reply:
I think we can just move this log inside the if statement because the actual set_control_type() mentioned in this log is performed only in that case.